mlos_bench/mlos_bench/launcher.py

```diff
@@ -129,16 +129,17 @@ def _parse_args(parser: argparse.ArgumentParser) -> Tuple[argparse.Namespace, Li
                  ' for additional config examples for this and other arguments.')
 
         parser.add_argument(
-            '--log_file', required=False,
+            '--log_file', '--log-file', required=False,
```

Comment:
I don't like mixing `-` and `_` in the option name, but the idea is to keep it consistent across the config parameters, internal argparse args names, and command-line options. Any config parameter can be a command line option, so after your change we'll end up with command lines like `--experiment_id foo --log-level DEBUG` and it would be hard to explain to the user why we can't have `--experiment-id` instead of `--experiment_id` but it's ok for `--log-level`

Reply:
so I would keep underscores everywhere for consistency, although I totally agree that this is not the most aesthetically pleasing solution

Reply:
I kept mistyping them locally, so I added aliases.  I'd like to keep them if possible.  I don't think they should add too much confusion - it's just in the cli args.

`--experiment-id` should probably move to the `globals` tbh, or we could just special treat it and convert.

Reply:
Part of the trouble as I see it is that we accept unknown arguments as globals, so if you accidentally use `--config-path` and we only support `--config_path`, then we get a global of `'config-path'` populated instead of setting the config paths or an error as one might expect.